packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx

```diff
@@ -202,28 +205,37 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
   const headerProps = useAdaptedSelector(getHeaderProps);
   const errorBarProps = usePropsFor(ErrorBar);
 
+  const overlayImageItemRef = React.useRef<OverlayImageItem | undefined>(overlayImageItem);
+  overlayImageItemRef.current = overlayImageItem;
+  const adapterChatMessages = useSelector(getChatMessages);
   useEffect(() => {
-    if (overlayImageItem === undefined) {
+    if (overlayImageItemRef.current === undefined) {
       return;
     }
-    const message = adapter.getState().thread.chatMessages[overlayImageItem.messageId];
+    const message: ChatMessageWithStatus | undefined = adapterChatMessages[overlayImageItemRef.current.messageId];
     if (message === undefined) {
       return;
     }
     const resourceCache = message.resourceCache;
-    if (overlayImageItem.imageSrc === '' && resourceCache && resourceCache[overlayImageItem.imageUrl]) {
-      const fullSizeImageSrc = getResourceSourceUrl(resourceCache[overlayImageItem.imageUrl]);
-      if (fullSizeImageSrc === undefined || fullSizeImageSrc === '' || overlayImageItem.imageSrc === fullSizeImageSrc) {
+    if (
+      overlayImageItemRef.current.imageSrc === '' &&
+      resourceCache &&
+      resourceCache[overlayImageItemRef.current.imageUrl]
+    ) {
+      const fullSizeImageSrc = getResourceSourceUrl(resourceCache[overlayImageItemRef.current.imageUrl]);
+      if (
+        fullSizeImageSrc === undefined ||
+        fullSizeImageSrc === '' ||
+        overlayImageItemRef.current.imageSrc === fullSizeImageSrc
+      ) {
         return;
       }
       setOverlayImageItem({
-        ...overlayImageItem,
+        ...overlayImageItemRef.current,
         imageSrc: fullSizeImageSrc
       });
     }
-    // Disable eslint because we are using the overlayImageItem in this effect but don't want to have it as a dependency, as it will cause an infinite loop.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [messageThreadProps.messages]);
+  }, [adapterChatMessages, messageThreadProps.messages]);
```

Comment:
it looks like `messageThreadProps.messages` is not used anymore in this `useEffect`

Reply:
Updated